src/ServiceControl/SagaAudit/SagaAuditing.cs

```diff
@@ -35,11 +35,26 @@ public override void Enrich(IReadOnlyDictionary<string, string> headers, IDictio
 
                         foreach (var part in multiSagaChanges.Select(s => s.Split(':')))
                         {
-                            sagasChanges.Add(part[0], part[1]);
+                            var id = part[0];
+                            var thisChange = part[1];
+                            string previousChange;
+                            if (!sagasChanges.TryGetValue(id, out previousChange))
+                            {
+                                sagasChanges[id] = thisChange;
+                            }
+                            else //Completed overrides New overrides Updated
+                            {
+                                if (string.Equals(thisChange, "Completed", StringComparison.OrdinalIgnoreCase)
```

Comment:
The plugin only ever generates these values with this case. Do we need the ignore case?

Some alternatives:
- Weirdly the values we care about are in alphabetical order. We could just collect them all and sort alphabetically and then return the first one
- We could parse the value into an enum and then sort them numerically

- If we go with a collect and sort later approach this whole thing could be replaced with an ILookup

```csharp
var values = Enumerable.Empty<string[]>();
if...
  values = multiSagaChanges.Select(s => s.Split(':'));
var lookup = values.ToLookup(v => v.FirstOrDefault(), v => v.Skip(1).FirstOrDefault());

// ...
.Select(saga => 
  change = lookup[sagaInvoked[1]].OrderBy(x => x).FirstOrDefault();
```